### PR TITLE
Upgrading slate to current version

### LINKED ISSUE #202: upgrading slate.js
hopefully, slate.js has a lot of bug fixed and we can improve our editor

- upgrade dependency
- check for breaking changes

### PR DESCRIPTION
**Pull request by Entkenntnis**
closes #202

### CODE REVIEW

File: packages/plugin-text/src/factory/editor.tsx [hunk 1]
```diff
@@ -167,7 +167,7 @@ export const createTextEditor = (
     const onClick = React.useCallback((e, editor, next): CoreEditor | void => {
       if (e.target) {
         // @ts-ignore
-        const node = findNode(e.target as Element, editor)
+        const node = editor.findNode(e.target as Element)
```
Comment (Knorrke) [inline packages/plugin-text/src/factory/editor.tsx]: `editor.current`?

Comment (Entkenntnis) [reply packages/plugin-text/src/factory/editor.tsx]: see https://github.com/ianstormtaylor/slate/blob/master/packages/slate-react/Changelog.md, 0.22 deprecated

Comment (inyono) [reply packages/plugin-text/src/factory/editor.tsx]: @Knorrke That's not the editor ref that's used further down. It's the second argument to `onClick`

**Comment by Entkenntnis** [comment]
it's working so far, may need some more testing

**Pull request merged**